presto-tests/src/test/java/com/facebook/presto/execution/TestConcurrentQueriesInSingleTransaction.java

```diff
@@ -0,0 +1,75 @@
+ * limitations under the License.
+ */
+package com.facebook.presto.execution;
+
+import com.facebook.presto.Session;
+import com.facebook.presto.resourceGroups.ResourceGroupManagerPlugin;
+import com.facebook.presto.spi.QueryId;
+import com.facebook.presto.tests.DistributedQueryRunner;
+import com.facebook.presto.transaction.TransactionId;
+import com.google.common.collect.ImmutableMap;
+import org.testng.annotations.Test;
+
+import static com.facebook.presto.execution.QueryState.FAILED;
+import static com.facebook.presto.execution.QueryState.FINISHED;
+import static com.facebook.presto.execution.QueryState.RUNNING;
+import static com.facebook.presto.execution.TestQueryRunnerUtil.LONG_LASTING_QUERY;
+import static com.facebook.presto.execution.TestQueryRunnerUtil.START_TXN_QUERY;
+import static com.facebook.presto.spi.ErrorType.USER_ERROR;
+import static com.facebook.presto.spi.StandardErrorCode.INVALID_CONCURRENT_QUERY;
+import static org.testng.Assert.assertEquals;
+import static org.testng.Assert.assertNotNull;
+
+public class TestConcurrentQueriesInSingleTransaction
+{
+    @Test(timeOut = 240_000)
+    public void testConcurrentQueryInSingleTransaction()
+            throws Exception
+    {
+        try (DistributedQueryRunner queryRunner = TestQueryRunnerUtil.createQueryRunner(ImmutableMap.of())) {
+            queryRunner.installPlugin(new ResourceGroupManagerPlugin());
+
+            Session clientSession = TestQueryRunnerUtil.newTransactionSession();
+
+            // start transaction
+            QueryId startTxnQry = TestQueryRunnerUtil.createQuery(queryRunner, clientSession, START_TXN_QUERY);
+
+            // wait until the transaction is registered
+            TestQueryRunnerUtil.waitForQueryState(queryRunner, startTxnQry, FINISHED);
+
+            assertEquals(queryRunner.getTransactionManager().getAllTransactionInfos().size(), 1);
+            TransactionId txnId = queryRunner.getTransactionManager().getAllTransactionInfos().get(0).getTransactionId();
+
+            // submit first "cli" query
+            QueryId firstCliQuery = TestQueryRunnerUtil.createQuery(queryRunner, clientSession, txnId, LONG_LASTING_QUERY);
+
+            // wait for the first query to start
+            TestQueryRunnerUtil.waitForQueryState(queryRunner, firstCliQuery, RUNNING);
+
+            // submit second "cli" query without waiting
+            QueryId secondCliQuery = TestQueryRunnerUtil.createQuery(queryRunner, clientSession, txnId, LONG_LASTING_QUERY);
+
+            // The second query should failed
+            QueryInfo failedQueryInfo = queryRunner.getCoordinator().getQueryManager().getQueryInfo(secondCliQuery);
+            assertNotNull(failedQueryInfo.getErrorType());
+            assertNotNull(failedQueryInfo.getErrorCode());
+            assertEquals(failedQueryInfo.getErrorType(), USER_ERROR);
+            assertEquals(failedQueryInfo.getErrorCode(), INVALID_CONCURRENT_QUERY.toErrorCode());
+
+            // Cancel the first query
```

Comment:
If we do it automatically at shutdown, there will be several stack traces in the test log because of some sever exceptions. Adding these two lines simplifies the test log and makes it easier to debug when the test fails.